File: src/aawireless/configuration/Configuration.cpp
//
// Created by chiel on 29-12-19.
//

#include <boost/property_tree/ptree.hpp>
#include <boost/property_tree/ini_parser.hpp>
#include "Configuration.h"

namespace aawireless {
    namespace configuration {
        Configuration::Configuration(const std::string &file) {
            boost::property_tree::ptree iniConfig;
            boost::property_tree::ini_parser::read_ini(file, iniConfig);

            wifiIpAddress = iniConfig.get<std::string>("Wifi.IpAddress");
            wifiPort = iniConfig.get<std::uint16_t>("Wifi.Port");
            wifiBSSID = iniConfig.get<std::string>("Wifi.BSSID");
            wifiSSID = iniConfig.get<std::string>("Wifi.SSID");
            wifiPassphrase = iniConfig.get<std::string>("Wifi.Passphrase");
        }
    }
}

File: src/aawireless/configuration/Configuration.rs
use boost;
use std;

// #include <boost/property_tree/ptree.hpp>
// #include <boost/property_tree/ini_parser.hpp>
// #include "Configuration.h"

pub struct Configuration {
    pub wifiDevice: String,
    pub wifiIpAddress: String,
    pub wifiPort: u16,
    pub wifiBSSID: String,
    pub wifiSSID: String,
    pub wifiPassphrase: String,
    wifiHotspotScript: String,
}
impl Configuration {
    pub fn new(&mut self, &file: String) -> Self {
        let iniConfig = boost::property_tree::ptree;
        boost::property_tree::ini_parser::read_ini(file, iniConfig);
    
        return Self {
            wifiDevice: iniConfig.get<String>("Wifi.Device"),
            wifiIpAddress: iniConfig.get<String>("Wifi.IpAddress"),
            wifiPort: iniConfig.get<u16>("Wifi.Port"),
            wifiSSID: iniConfig.get<String>("Wifi.SSID"),
            wifiBSSID: iniConfig.get<String>("Wifi.BSSID"),
            wifiPassphrase: iniConfig.get<String>("Wifi.Passphrase"),
            wifiHotspotScript: iniConfig.get<String>("Wifi.HotspotScript"),
        }
        //wifiPassphrase: iniConfig.get<String>("Wifi.Passphrase"); // TODO: implement for hardcoded passphrase instead of generated
    }
}
